cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueAdmin.java

```diff
@@ -482,60 +479,51 @@ public void configureGroups(QueueName queueName, Map<Long, Integer> groupInfo) t
 
   @Override
   public void upgrade() throws Exception {
-    // For each table managed by this admin, performs an upgrade
-    for (HTableDescriptor desc : getHBaseAdmin().listTables()) {
-      String tableName = Bytes.toString(desc.getName());
+    // For each table managed by this admin, perform an upgrade
+    List<TableId> tableIds = tableUtil.listTables(getHBaseAdmin());
+    for (TableId tableId : tableIds) {
       // It's important to skip config table enabled.
-      if (isDataTable(tableName)) {
-        LOG.info(String.format("Upgrading queue hbase table: %s", tableName));
+      if (isDataTable(tableId)) {
+        LOG.info(String.format("Upgrading queue table: %s", tableId));
         Properties properties = new Properties();
+        HTableDescriptor desc = tableUtil.getHTableDescriptor(getHBaseAdmin(), tableId);
         if (desc.getValue(HBaseQueueAdmin.PROPERTY_PREFIX_BYTES) == null) {
           // It's the old queue table. Set the property prefix bytes to SALT_BYTES
           properties.setProperty(HBaseQueueAdmin.PROPERTY_PREFIX_BYTES, Integer.toString(HBaseQueueAdmin.SALT_BYTES));
         }
-        upgrade(tableName, properties);
-        LOG.info(String.format("Upgraded queue hbase table: %s", tableName));
+        upgrade(tableId, properties);
+        LOG.info(String.format("Upgraded queue table: %s", tableId));
       }
     }
   }
 
+  private void upgrade(TableId tableId, Properties properties) throws Exception {
+    AbstractHBaseDataSetAdmin dsAdmin = new DatasetAdmin(tableId, hConf, tableUtil, properties);
+    try {
+      dsAdmin.upgrade();
+    } finally {
+      dsAdmin.close();
+    }
+  }
+
   /**
-   * @param tableName being checked
+   * @param tableId TableId being checked
    * @return true if the given table is the actual table for the queue (opposed to the config table for the queue
    * or tables for things other than queues).
    */
-  private boolean isDataTable(String tableName) {
-    // checks if table is constructed by getActualTableName(String)
+  private boolean isDataTable(TableId tableId) {
+    // checks if table is constructed by getDataTableId
+    String tableName = tableId.getTableName();
     String[] parts = tableName.split("\\.");
-    if (parts.length != 6) {
+    if (parts.length != 4) {
```

Comment:
You mean `if (!isConfigTable(tableName)`?

I don't want to upgrade all other tables, simply because its not a config table (could be *any* other table created by cdap).